httpx_ws/transport.py

```diff
@@ -39,26 +39,36 @@ def __init__(self, app: ASGIApp, scope: Scope) -> None:
         self._send_queue: queue.Queue[Message] = queue.Queue()
         self.connection = wsproto.WSConnection(wsproto.ConnectionType.SERVER)
         self.connection.initiate_upgrade_connection(scope["headers"], scope["path"])
+        self._exit_stack = contextlib.AsyncExitStack()
+        self._aentered = False
 
     async def __aenter__(
         self,
     ) -> tuple["ASGIWebSocketAsyncNetworkStream", bytes]:
-        self._task_group = await anyio.create_task_group().__aenter__()
-        self._task_group.start_soon(self._run)
+        if self._aentered:
+            raise RuntimeError(
+                "Cannot use ASGIWebSocketAsyncNetworkStream in a context manager twice"
+            )
+        self._aentered = True
+        async with contextlib.AsyncExitStack() as stack:
+            self._task_group = await stack.enter_async_context(anyio.create_task_group())
+            self._task_group.start_soon(self._run)
+
+            await self.send({"type": "websocket.connect"})
```

Comment:
if send was cancelled it would not propagate into the TaskGroup async context manager!

This was fixed by using two AsyncExitStacks and pop_all(), so all code is covered by a context manager and therefore the TaskGroup.